src/domain/routes/index.types.d.ts

```diff
@@ -0,0 +1,58 @@
+type RawHop = {
+  /** Standard Format Channel Id */
+  channel: string
+  /** Channel Capacity Tokens */
+  channel_capacity: number
+  /** Fee */
+  fee: number
+  /** Fee Millitokens */
+  fee_mtokens: string
+  /** Forward Tokens */
+  forward: number
+  /** Forward Millitokens */
+  forward_mtokens: string
+  /** Public Key Hex */
+  public_key?: string
+  /** Timeout Block Height */
+  timeout: number
+}
+
+type RawRoute = {
```

Comment:
#### Open question (cc @bodymindarts)

I'm having problems reconciling this type. The `RawRoute` type (I made up):
- first gets received from calling `lnService.probeForRoute` (no in-code type found, documented types [here](https://github.com/alexbosworth/ln-service#probeForRoute))
- gets persisted to redis cache along with the pubkey (this becomes `CachedRoute` defined [lower down here](https://github.com/GaloyMoney/galoy/pull/594/files#diff-89760a3bcdcc20c6dd12e0ffd947b3dccccb9b75164c1da95eef1e3a2f604edeR40) by us)
- gets read from cache as part of the `CachedRoute` type
- eventually gets passed back to the `payViaRoutes` method which expects a slightly different type (defined in-code [here](https://github.com/alexbosworth/lightning/blob/9a4d0c726a6dbe455108bc8cb80e8b6cfb47d9c6/lnd_methods/offchain/pay_via_routes.d.ts#L12), documented [here](https://github.com/alexbosworth/ln-service#payViaRoutes))

The type is inconsistent when comparing the `probeForRoute` return and the `payViaRoutes` arg, but the arg is a subset of the return type since we pass the return value directly to the `payViaRoutes` method via the redis cache in the existing codebase.

The type also exists as part of other types when it exists in the library, but not separately as itself.

**The question:** is it ok what I'm trying to do here by trying to isolate the smallest subset of common things from the `probeForRoute` return to pass forward to the `payViaRoutes` method, even though it breaks some of our naming conventions? Do we need some sort of translation for this instead (I suspect this is the case)?

Reply:
maybe we should update the package, I had to do it for admin queries. you are referencing the last code/docs from https://github.com/alexbosworth/ln-service#probeForRoute or did you check previous versions?

Reply:
I was able to reconcile this by making the types [like this](https://github.com/GaloyMoney/galoy/pull/594/files#diff-89760a3bcdcc20c6dd12e0ffd947b3dccccb9b75164c1da95eef1e3a2f604edeR1)

I was thinking about doing a PR, but I'm not sure I'd push this back up to the library because it may not be a clean change there 🤔